addons/l10n_th/models/res_partner.py

```diff
@@ -0,0 +1,12 @@
+from odoo import models
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _l10n_th_get_company_info(self, use_code=False):
+        code = self.company_registry or ""
+        if use_code:
+            return code or ""
+        if not self.company_registry:
+            return ""
+        return "Headquarter" if code == "0000" else "Branch " + code
```

Comment:
I think for this, there could be possibility that the invoice is sent to an individual instead of a company. I will remove the use_code but I think returning empty string ("") is important.